docs/src/modules/shallowwater.md

```diff
@@ -0,0 +1,17 @@
+# ShallowWater Module
+
+### Basic Equations
+
+This module solves the 2D shallow water equations:
+
+```math
+\begin{aligned}
+\frac{\partial u}{\partial t} + \boldsymbol{u \cdot \nabla} u - f v & = - g \frac{\partial \eta}{\partial x} - \mathrm{D} u, \\
+\frac{\partial v}{\partial t} + \boldsymbol{u \cdot \nabla} v + f u & = - \mathrm{D} v, \\
+\frac{\partial \eta}{\partial t} + \boldsymbol{\nabla \cdot} ( \boldsymbol{u} h ) & = - \mathrm{D} \eta.
```

Comment:
Should we write in conservative form?

Reply:
1) Yes. 2) Also there are some typos, e.g., the `-g ∂η/∂y` term is missing from the `v` equation. 3) Also I'd like to include non-trivial bathymetry (which appears as a forcing term on the right-hand side in the momentum equations.)

Reply:
Here they are in conservative form:

<img width="833" alt="Screen Shot 2021-01-11 at 1 51 32 pm" src="https://user-images.githubusercontent.com/7112768/104143610-51991c80-5414-11eb-8703-cf82c2ec79e8.png">

I made an attempt to write the `calcN!` function. Bit tricky...

Also, what's the form of dissipation? We want it to conserve the total `hu`, `hv` and `h`... Right?

Reply:
We're having a related discussion at https://github.com/CliMA/Oceananigans.jl/issues/1712